qt/scientific_interfaces/Indirect/ISISEnergyTransfer.cpp

```diff
@@ -92,6 +98,54 @@ void ungroupWorkspace(std::string const &workspaceName) {
   ungroup->execute();
 }
 
+IAlgorithm_sptr loadAlgorithm(std::string const &filename,
+                              std::string const &outputName) {
+  auto loader = AlgorithmManager::Instance().create("Load");
+  loader->initialize();
+  loader->setProperty("Filename", filename);
+  loader->setProperty("OutputWorkspace", outputName);
+  return loader;
+}
+
+void deleteWorkspace(std::string const &name) {
+  auto deleter = AlgorithmManager::Instance().create("DeleteWorkspace");
+  deleter->initialize();
+  deleter->setProperty("Workspace", name);
+  deleter->execute();
+}
+
+double getSampleLog(std::string const &workspaceName,
+                    std::string const &logName, double const &defaultValue) {
+  try {
+    return getADSMatrixWorkspace(workspaceName)->getLogAsSingleValue(logName);
+  } catch (std::exception const &) {
+    return defaultValue;
+  }
+}
+
+double getSampleLog(std::string const &workspaceName,
+                    std::vector<std::string> const &logNames,
+                    double const &defaultValue) {
+  double value(defaultValue);
+  for (auto const &logName : logNames) {
+    value = getSampleLog(workspaceName, logName, defaultValue);
+    if (value != defaultValue)
+      break;
+  }
+  deleteWorkspace(workspaceName);
+  return value;
+}
+
+double loadSampleLog(std::string const &filename,
+                     std::vector<std::string> const &logNames,
+                     double const &defaultValue) {
+  auto const temporaryWorkspace("__sample_log_subject");
+  auto loader = loadAlgorithm(filename, temporaryWorkspace);
+  loader->execute();
```

Comment:
Similarly, can you not get the outputworkspace property from the loader, and get sample log from this workspace, rather than create an ADS instance and retrieve the workspace

Reply:
`loader->getProperty("OutputWorkspace")` will only return the name of the output workspace in this case because the algorithm is not a child. I could make the load algorithm a child, however I believe it is better suited being in the anonymous namespace.